CppCoreGuidelines.md

```diff
@@ -5968,26 +5969,30 @@ You can handle self-assignment by explicitly testing for self-assignment, but of
 
     Foo& Foo::operator=(const Foo& a)   // OK, but there is a cost
     {
-        if (this == &a) return *this;
-        s = a.s;
-        i = a.i;
+        if (this != &a) {
+            s = a.s;
+            i = a.i;
+        }
         return *this;
     }
 
-This is obviously safe and apparently efficient.
-However, what if we do one self-assignment per million assignments?
-That's about a million redundant tests (but since the answer is essentially always the same, the computer's branch predictor will guess right essentially every time).
-Consider:
+The code we are skipping with the test would actually work fine even for self-assignment;
```

Comment:
the argument in the deleted text was about general case, where the test is needed for correctness. The example was poorly chosen.